src/dom.js

```diff
@@ -203,6 +203,42 @@ export function childElement(parent, callback) {
   return null;
 }
 
+
+/**
+ * Finds all child elements that satisfies the callback.
+ * @param {!Element} parent
+ * @param {function(!Element):boolean} callback
+ * @return {!Array.<!Element>}
+ */
+export function childElements(parent, callback) {
```

Comment:
There are a few places that can use this, like `CustomElement#getRealChildren`. Can you open an issue to updates those uses?